tika-parsers/src/main/java/org/apache/tika/parser/tagRatio/TextToTagRatio.java

```diff
@@ -0,0 +1,209 @@
+            StringBuffer sb = new StringBuffer();
+            double tagRatio = 0.0;
+            int count = 0;
+            br = new BufferedReader(new StringReader(nl.toHtml()));
+            for (int i = 0; i < linkTagList.length; i++) {
+                line = br.readLine();
+                line = line.trim();
+                if (line.equals("")) {
+                    continue;
+                }
+                linkTagList[i] = computeTextToTagRatio(line);
+                //Extract meta tags
+                populateMetaTags(line, metaTagsMap);
+                
+                if(linkTagList[i] != 0 && linkTagList[i] >= threshold){
+                    
+                    sb.append(line);
+                    sb.append("\n");
+                    tagRatio += linkTagList[i];
+                    count++;
+                }
+            }
+            
+            //Create a JSON Object
+            JSONObject obj = new JSONObject();
+            obj.put("avgTagRatio", (tagRatio/count));
+            obj.put("content", sb.toString());
+            
+            JSONArray array = new JSONArray();
+            array.put(metaTagsMap);
+            obj.put("meta-tags", array);
+            
+            
+            br.close();
+            
+        }catch (MalformedURLException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (ParserException e) {
+            e.printStackTrace();
+            return linkTagList;
+        }
+        return linkTagList;
+    }
+    
+    /**
+     * Populate the metadata information
+     * @param line
+     * @param metaTagsMap Hash Map to store metadata as key value pairs
+     * @return
+     */
+    
+    private void populateMetaTags(String line, HashMap<String, String> metaTagsMap) {
+        
+        if(line.startsWith("<meta ")){
+            
+            Document doc = Jsoup.parse(line);
+            Element tag = doc.select("meta").first();
+            String name = tag.attr("name");
+            String content = tag.attr("content");
+            
+            metaTagsMap.put(name, content);
+        }
+    }
+    
+    /**
+     * Calculate Text to Tag Ratio for line
+     * @param line Line in the raw html
+     * @return Tag Ratio of the line
+     */
+    private double computeTextToTagRatio(String line) {
+        int tag = 0;
+        int text = 0;
+        
+        for (int i = 0; i >= 0 && i < line.length(); i++) {
+            if (line.charAt(i) == '<') {	//start tag
+                tag++;
+                i = line.indexOf('>', i);
+                if (i == -1) {
+                    break;
+                }
+            } else if (tag == 0 && line.charAt(i) == '>') {	//close tag
+                text = 0;
+                tag++;
+            } else {		//just text
+                text++;
+            }
+            
+        }
+        if (tag == 0) {
+            tag = 1;
+        }
```

Comment:
Yes tag is not supposed to be 0 since we are computing text/tag ratio. To avoid the denominator value from becoming zero while computing the ratio, I added that change. This logic also makes sense and like you said its more readable as well.